HW05.Rmd

```diff
@@ -0,0 +1,166 @@
+---
+title: "HW05"
+output: github_document
+author: "Daniel Krochmal"
+date: "8/4/2020"
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+``` {r load libraries, echo = FALSE, message = FALSE}
+library(readr)
+library(knitr)
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+```
+
+
+## Part 1
+
+
+``` {r import data, echo = FALSE, message = FALSE}
+Dad_mom <- read_tsv("dad_mom.txt") #Importing data using readr
+```
+
+``` {r tidy data}
+
+#I will use a sequence of gather(), separate() and spread() verbs to clean-up the datafame
+
+tidyDad_mom <- Dad_mom %>%
+  
+  gather(key = "var", value = "value",
+         convert = FALSE, -fam_id) %>%
+  
+  #Since the "var" column contains two variables, I'm separating them into "var" (name,income) and "relation" (mom,dad)
+  separate(col = var, into = c("var", "relation")) %>%
+  
+  #Since name and income should have their own column, I'm using spread() to accomplish that 
+  spread(key = var, value = value, convert = TRUE) %>%
+  
+  #Reordering column using select. I wanted the "name" to be a second column, so for the rest I used everything()
+  select(fam_id, name, everything()) %>%
+  rename(family = fam_id)
+
+kable(
+  tidyDad_mom,
+  format = "markdown",
+  align = "c"
+)
+```
+## Part 2
+
+``` {r import files, echo = FALSE, message = FALSE}
+
+#importing csv files using readr
+auc1 <- read_csv("CTRP_files/AUC_1.csv")
+auc2 <- read_csv("CTRP_files/AUC_2.csv")
+ccl <- read_csv("CTRP_files/cancer_cell_line_info.csv")
+cpd <- read_csv("CTRP_files/compound_info.csv")
+exp <- read_csv("CTRP_files/Experiment_info.csv")
+
+```
+``` {r join data, echo = FALSE}
+ctrp_data <- auc1 %>%
+  
+  #Binding second part of the experimental data (auc2) to auc1
+  bind_rows(auc2) %>%
+  
+  #Adding experiment info to the experimental data
+  left_join(exp, by = c("experiment_id" = "expt_id")) %>%
+  
+  #Adding compound info
+  left_join(cpd, by = "master_cpd_id" ) %>%
+  
+  #Adding cell line info
+  left_join(ccl, by = "master_ccl_id")
```

Comment:
It was smart to call `left_join()`. I had called `inner_join()` but, given the nature of the dataset, `left_join()` is most appropriate.